x/ccv/provider/keeper/throttle.go

```diff
@@ -321,6 +322,43 @@ func (k Keeper) QueueThrottledPacketData(
 	k.IncrementThrottledPacketDataSize(ctx, consumerChainID)
 }
 
+// GetLeadingVSCMaturedData returns the leading vsc matured packet data instances
+// for a chain-specific throttled packet data queue. Ie the vsc matured packet data instances
+// that do not have any slash packet data instances preceding them in the queue for consumerChainID.
+func (k Keeper) GetLeadingVSCMaturedData(ctx sdktypes.Context, consumerChainID string) (
+	vscMaturedData []ccvtypes.VSCMaturedPacketData, ibcSeqNums []uint64) {
+
+	store := ctx.KVStore(k.storeKey)
+	iteratorPrefix := providertypes.ChainIdWithLenKey(providertypes.ThrottledPacketDataBytePrefix, consumerChainID)
+	iterator := sdktypes.KVStorePrefixIterator(store, iteratorPrefix)
+	defer iterator.Close()
+
+	// Iterate over the throttled packet data queue,
+	// and return vsc matured packet data instances until we encounter a slash packet data instance.
+	vscMaturedData = []ccvtypes.VSCMaturedPacketData{}
+	ibcSeqNums = []uint64{}
+	for ; iterator.Valid(); iterator.Next() {
+
+		bz := iterator.Value()
+		if bz[0] == slashPacketData {
```

Comment:
Nit:
Maybe use a switch in this case since all processing happens only for vscs, slash packet causes a break and an unknown type panics.

```golang
bz := iterator.Value()
switch bz[0] {
case vscMaturedPacketData:
	var data ccvtypes.VSCMaturedPacketData
	err := data.Unmarshal(bz[1:])
	if err != nil {
		panic(fmt.Sprintf("failed to unmarshal vsc matured packet data: %v", err))
	}
	
	vscMaturedData = append(vscMaturedData, data)
	_, ibcSeqNum := providertypes.MustParseThrottledPacketDataKey(iterator.Key())
	ibcSeqNums = append(ibcSeqNums, ibcSeqNum)
case slashPacketData:
	break
default:
	panic(fmt.Sprintf("unexpected packet data type: %d", bz[0]))
}
```

Reply:
This does look more readable, however the `break` you've defined in this code would break out of the switch case, and not the for-loop. Therefore we'd either need a break label, or the current code which uses if/else. 

A recent PR of mine had a break label, but it was decided from reviews that the pattern is not desirable